src/material-experimental/mdc-button/button-base.ts

```diff
@@ -78,6 +79,10 @@ export const _MatButtonBaseMixin: CanDisableRippleCtor&CanDisableCtor&CanColorCt
     typeof MatButtonMixinCore = mixinColor(mixinDisabled(mixinDisableRipple(MatButtonMixinCore)));
 
 /** Base class for all buttons.  */
+@Directive({
+  // TODO(devversion): this selector can be removed when we update to Angular 9.0.
+  selector: 'do-not-use-abstract-mat-button-base'
```

Comment:
Should the `abstract-` part of this (and others in this PR) be removed since this class isn't abstract?

Reply:
I think it's correct to call them abstract directives. That seems to be the official term for directives which are not intended to be used in templates (based on the commit messages on framework that introduced these)